packages/terra-table/src/Table.jsx

```diff
@@ -427,28 +436,49 @@ function Table(props) {
   // useEffect to calculate pinned column offsets
   useEffect(() => {
     const offsetArray = [];
+    const cellOffsetArray = [];
     let cumulativeOffset = 0;
+    let cellCumulativeOffset = 0;
     let lastPinnedColumnIndex;
 
     // if table has selectable rows but no pinned columns, then set the offset of the first column to 0
     if (hasSelectableRows && pinnedColumns.length === 0) {
       lastPinnedColumnIndex = 0;
       offsetArray.push(cumulativeOffset);
       setPinnedColumnOffsets(offsetArray);
+      setPinnedColumnHeaderOffsets(offsetArray);
       return;
     }
 
     if (pinnedColumns.length > 0) {
       offsetArray.push(cumulativeOffset);
-
+      cellOffsetArray.push(cellCumulativeOffset);
       lastPinnedColumnIndex = hasSelectableRows ? pinnedColumns.length : pinnedColumns.length - 1;
 
+      if (pinnedColumns[0].columnSpan > 1) {
+        for (let c = pinnedColumns[0].columnSpan; c > 1; c -= 1) {
```

Comment:
We should have a more meaningful iterator variable name than "c".  Does "c" stand for column?